src/Middleware/HttpLogging/src/HttpLoggingInterceptorContext.cs

```diff
@@ -0,0 +1,147 @@
+    public HttpLoggingFields LoggingFields { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of bytes of the request body to log.
+    /// </summary>
+    /// <remarks>
+    /// This is pre-populated with the value from <see cref="HttpLoggingOptions.RequestBodyLogLimit"/>,
+    /// <see cref="HttpLoggingAttribute.RequestBodyLogLimit"/>, or
+    /// <see cref="HttpLoggingEndpointConventionBuilderExtensions.WithHttpLogging{TBuilder}(TBuilder, HttpLoggingFields, int?, int?)"/>.
+    /// </remarks>
+    public int RequestBodyLogLimit { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum number of bytes of the response body to log.
+    /// </summary>
+    /// <remarks>
+    /// This is pre-populated with the value from <see cref="HttpLoggingOptions.ResponseBodyLogLimit"/>,
+    /// <see cref="HttpLoggingAttribute.ResponseBodyLogLimit"/>, or
+    /// <see cref="HttpLoggingEndpointConventionBuilderExtensions.WithHttpLogging{TBuilder}(TBuilder, HttpLoggingFields, int?, int?)"/>.
+    /// </remarks>
+    public int ResponseBodyLogLimit { get; set; }
+
+    internal long StartTimestamp { get; set; }
+    internal TimeProvider TimeProvider { get; set; } = null!;
+
+    /// <summary>
+    /// Data that will be logged as part of the request or response. Values specified in <see cref="LoggingFields"/>
+    /// will be added automatically after all interceptors run. These values are cleared after logging the request.
+    /// All other relevant settings will carry over to the response.
+    /// </summary>
+    public IList<KeyValuePair<string, object?>> Parameters { get; } = new List<KeyValuePair<string, object?>>();
+
+    /// <summary>
+    /// Adds data that will be logged as part of the request or response. See <see cref="Parameters"/>.
+    /// </summary>
+    /// <param name="key">The parameter name.</param>
+    /// <param name="value">The parameter value.</param>
+    public void AddParameter(string key, object? value)
+    {
+        Parameters.Add(new(key, value));
+    }
+
+    /// <summary>
+    /// Adds the given fields to what's currently enabled in <see cref="LoggingFields"/>.
+    /// </summary>
+    /// <param name="fields"></param>
+    public void Enable(HttpLoggingFields fields)
+    {
+        LoggingFields |= fields;
+    }
+
+    /// <summary>
+    /// Checks if any of the given fields are currently enabled in <see cref="LoggingFields"/>.
+    /// </summary>
+    public bool IsAnyEnabled(HttpLoggingFields fields)
+    {
+        return (LoggingFields & fields) != HttpLoggingFields.None;
+    }
+
+    /// <summary>
+    /// Removes the given fields from what's currently enabled in <see cref="LoggingFields"/>.
+    /// </summary>
+    /// <param name="fields"></param>
+    public void Disable(HttpLoggingFields fields)
+    {
+        LoggingFields &= ~fields;
+    }
+
+    /// <summary>
+    /// Checks if any of the given fields are currently enabled in <see cref="LoggingFields"/>
```

Comment:
IMO, rewrite the comment to focus on disabling. e.g. "Disable the specified fields if they are enabled."